### ISSUE TITLE
Implement XYZ order in quaternion.toEuler

### ISSUE THREAD

**Issue by SET001** [schteppe/cannon.js#385]
I have spend few crazy hours trying to understand what is wrong in my application. When you want to convert from Euler to Quaternion you have `quaternion.toEuler` and `quaternion.setFromEuler`. And while for the first one default order is `YZX`, this is not the same for inverse transformation - you have to pass `XYZ` in that case. This is very misleading.

Right now the error is `Euler order XYZ not supported yet.` 

The workaround is to use something like this:

`physic.rigidBody.quaternion.toEuler(new Vec3(rotation.y, rotation.z, rotation.x))`

instead of what would be easier and more intuitive:

`physic.rigidBody.quaternion.toEuler(rotation)`

I would create a pull request for this but I'm totally zero in math *(

**Comment by schteppe** [comment]
Yes, I agree that both should be the same order by default. Or maybe toEuler should be removed?
if you’re using quaternion to euler, in my experience something might be badly designed in your code. Some quaternions can yield multiple euler angles. This is why I’ve stopped using toEuler() code completely, in any context, and only use euler angles as input. Usually there is an alternative solution. How are you using euler in your application?

Just checked Bullet physics, and I can find fromEuler() equivalents, but no toEuler()... http://bulletphysics.org/Bullet/BulletFull/classbtQuaternion.html

If you still need toEuler, a workaround is to use your rendering library quaternion/euler math and then copy the result into the cannon.js quaternion. I know Three.js has good quaternion/euler support.

**Comment by SET001** [comment]
I have an Entity Component System architecture in my game and I have rotation component which had x, y and z properties. Render and Physic system should read from it and update. In physic system  I used quaternion to rotate things so I have to make transformation from euler to quaternion and backward  (using toEuler and  fromEuler). Now I switch rotation component to quaternion, as you offered and now I don't need those transformation any more. Everything works fine except that quaternions operations is a bit more complicated for some one without basic math knowledge.

**Comment by robrohan** [comment]
This video helped me, might help you as well 
https://youtu.be/SCbpxiCN0U0
